gotopkg.go

```diff
@@ -1,204 +0,0 @@
-			lastSince = time.Since(recipe.lastStarted)
-			if !recipe.firstRun || lastSince > RUN_EVERY {
-				recipes[i].firstRun = true
-				recipes[i].lastStarted = time.Now()
-				log.Printf("running recipe %s (first run or last run > %s [%s])", recipe.name, RUN_EVERY, lastSince)
-
-				autopkgRunOneRecipeCheckDLFirst(recipe.name, false)
-			}
-		}
-
-		log.Printf("sleeping %s", TIME_DELAY)
-		time.Sleep(TIME_DELAY)
-	}
-}
-
-func autopkgRunOneRecipeCheckDLFirst(recipe string, forceFull bool) {
-	var report AutopkgRunReport
-
-	// run, but only check first
-	if !forceFull {
-		output, err := autopkgRun([]string{"-v", "-c", recipe}, &report)
-		log.Printf("check run: %v", report)
-		if err != nil {
-			log.Printf("error: %s", err)
-			if output != nil {
-				log.Printf("%s", string(output[:]))
-			}
-			return
-		}
-		if output != nil {
-			log.Printf("check output: %s", string(output[:]))
-		}
-		if len(report.SummaryResults.URLDownloaderSummaryResult.DataRows) < 1 {
-			log.Printf("nothing downloaded, skipping full run")
-			return
-		}
-	}
-
-	report = AutopkgRunReport{}
-	cmdArgs := []string{"-v", recipe}
-	if strings.HasSuffix(recipe, "munki") {
-		cmdArgs = append(cmdArgs, "MakeCatalogs.munki")
-	}
-	output, err := autopkgRun(cmdArgs, &report)
-	if err != nil {
-		log.Printf("%s", err)
-	}
-	if output != nil {
-		log.Printf("run output: %s", string(output[:]))
-	}
-	if len(report.SummaryResults.MunkiImporterSummaryResult.DataRows) < 1 {
-		log.Printf("no munki items imported")
-		return
-	}
-
-	log.Printf("finished full autopkg run")
-}
-
-func autopkgRun(params []string, report *AutopkgRunReport) (output []byte, err error) {
-	tmpReport, err := ioutil.TempFile("", "gotopkg")
-	if err != nil {
-		return nil, fmt.Errorf("error creating report temp file: %v", err)
-	}
-	tmpReport.Close() // will be overwritten by autopkg anyway
-
-	cmdArgs := []string{"run", "--report-plist", tmpReport.Name()}
-	cmdArgs = append(cmdArgs, params...)
-
-	log.Printf("exec autopkg: %s %s", autopkg, strings.Join(cmdArgs, " "))
-	output, err = exec.Command(autopkg, cmdArgs...).CombinedOutput()
-	if err != nil {
-		if exiterr, ok := err.(*exec.ExitError); ok {
-			exitCode := exiterr.Sys().(syscall.WaitStatus).ExitStatus()
-			// autopkg exit code 70 (RECIPE_FAILED_CODE) means that an autopkg
-			// Processor has suffered an exception. because this could be for
-			// rather innocuous reasons (and that autopkg itself has caught
-			// the problem) we'll refrain from passing on this error. besides
-			// it will be included in the report plist, too
-			if exitCode != RECIPE_FAILED_CODE {
-				return output, fmt.Errorf("error executing autopkg (non-zero return %d): %v", exitCode, err)
-			}
-		} else {
-			return output, fmt.Errorf("error executing autopkg: %v", err)
-		}
-	}
-
-	f, err := os.Open(tmpReport.Name())
-	if err != nil {
-		return output, fmt.Errorf("error opening report temp file: %v", err)
-	}
-
-	err = report.DecodePlist(f)
-	if err != nil {
```

Comment:
Makes sense. I highly recommend https://github.com/pkg/errors for wrapping errors with some additional context.